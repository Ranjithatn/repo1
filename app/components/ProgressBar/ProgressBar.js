import React from 'react'

function ProgressBar({ value, max, type }) {
   return (
    <progress className={"progress " + type} value={value} max={max}></progress>
   )
}

ProgressBar.defaultProps = {
  max: "100",
  type: "is-primary"
}

export default ProgressBar;
