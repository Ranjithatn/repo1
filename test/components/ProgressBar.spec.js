import React from "react";
import toJson from "enzyme-to-json";
import ProgressBar from "../../app/components/ProgressBar/ProgressBar";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, render, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe( '<ProgressBar />', () => {

	it("should render without crashing", () => {
		const component = mount(
			<ProgressBar />
		);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});

	it( 'It should render with 0 value', () => {
		const wrapper = shallow( <ProgressBar /> );
		expect(wrapper.length).toBe(1);
	});


	it( 'component has class name `progess is-primary`', () => {
		const props = {
			max: 100,
			value: 40,
			type: 'is-primary',
		}
		const wrapper = mount( <ProgressBar {...props} /> );
		expect( wrapper.prop('value') ).toEqual(40);
	});


});

