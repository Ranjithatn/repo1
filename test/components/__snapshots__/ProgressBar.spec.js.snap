// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render without crashing 1`] = `
<ProgressBar
  max="100"
  type="is-primary"
>
  <progress
    className="progress is-primary"
    max="100"
  />
</ProgressBar>
`;
